### ISSUE TITLE
`CELERY_DEFAULT_DELIVERY_MODE` option doesn't work

### ISSUE THREAD

**Issue by tahajahangir** [celery/celery#1953]
Running the following script (after deleting already created queues/exchanges), creates a durable queue and a persistent message:

```python
import celery
app = celery.Celery()
app.config_from_object({'CELERY_DEFAULT_DELIVERY_MODE': 'transient'})

@app.task
def foo(): pass

foo.delay()

conn = app.broker_connection()
consumer = app.amqp.get_task_consumer(conn)
messages = []

def callback(body, message): messages.append(message)

consumer.register_callback(callback)
consumer.consume()
conn.drain_events(timeout=1)
print(messages[0].properties)
```
outputs:
```python
{'delivery_mode': 2, 'application_headers': {}, 'priority': 0, 'correlation_id': u'ecfe590c-de7a-46a4-88b5-7eab22133b3c', 'content_encoding': u'binary', 'content_type': u'application/x-python-serialize', 'reply_to': u'5d55e305-1fc4-326b-803a-43e21e759d8f'}
```

**Comment by pfreixes** [comment]
Is there any release date for the 3.2 version ?

**Comment by teuneboon** [comment]
sorry to bring this issue back from the grave. But am I correct in assuming the example provided here: http://docs.celeryproject.org/en/latest/userguide/optimizing.html#using-transient-queues is still wrong? It sets delivery_mode on the Queue, and according to one of @ask 's comments it should be set in Exchange.

**Comment by ask** [comment]
The issue is not fixed in 3.1, as the changes are quite involved, the fix only applies to 4.0dev:
http://docs.celeryproject.org/en/master/userguide/optimizing.html#using-transient-queues
